databuilder/databuilder/task/neo4j_staleness_removal_task.py

```diff
@@ -182,75 +210,84 @@ def _batch_delete(self,
+                                target_type: str
                                 ) -> None:
-        total_count_dict = {record['type']: int(record['count']) for record in total_records}
+        if total_record_count == 0 or stale_record_count == 0:
+            return
 
-        for record in stale_records:
-            type_str = record['type']
-            if type_str not in types:
-                continue
+        stale_pct = stale_record_count * 100 / total_record_count
 
-            stale_count = record['count']
-            if stale_count == 0:
-                continue
-
-            node_count = total_count_dict[type_str]
-            stale_pct = stale_count * 100 / node_count
-
-            threshold = self.staleness_pct_dict.get(type_str, self.staleness_pct)
-            if stale_pct >= threshold:
-                raise Exception(f'Staleness percentage of {type_str} is {stale_pct} %. '
-                                f'Stopping due to over threshold {threshold} %')
+        threshold = self.staleness_pct_dict.get(target_type, self.staleness_pct)
+        if stale_pct >= threshold:
+            raise Exception(f'Staleness percentage of {target_type} is {stale_pct} %. '
+                            f'Stopping due to over threshold {threshold} %')
 
     def _validate_node_staleness_pct(self) -> None:
-        total_nodes_statement = textwrap.dedent("""
-        MATCH (n)
-        WITH DISTINCT labels(n) as node, count(*) as count
-        RETURN head(node) as type, count
-        """)
-
-        stale_nodes_statement = textwrap.dedent("""
-        MATCH (n)
-        WHERE {}
-        WITH DISTINCT labels(n) as node, count(*) as count
-        RETURN head(node) as type, count
-        """)
-
-        stale_nodes_statement = textwrap.dedent(self._decorate_staleness(stale_nodes_statement))
-
-        total_records = self._execute_cypher_query(statement=total_nodes_statement)
-        stale_records = self._execute_cypher_query(statement=stale_nodes_statement,
-                                                   param_dict={MARKER_VAR_NAME: self.marker})
-        self._validate_staleness_pct(total_records=total_records,
-                                     stale_records=stale_records,
-                                     types=self.target_nodes)
+        total_nodes_statement = textwrap.dedent(
+            self.validate_node_staleness_statement.format(staleness_condition=''))
+        stale_nodes_statement = textwrap.dedent(
+            self._decorate_staleness(self.validate_node_staleness_statement))
+
+        for t in self.target_nodes:
+            if isinstance(t, TargetWithCondition):
+                target_type = t.target_type
+                total_extra_condition = 'WHERE ' + t.condition
+                stale_extra_condition = ' AND ' + t.condition
+            else:
+                target_type = t
+                total_extra_condition = ''
+                stale_extra_condition = ''
```

Comment:
i feel like this could be simpler if you always include the WHERE part (in the statement itself) an for the total statement you format the staleness condition as just "true". WDYT